grant-project/grant-monorepo
# Readable Proposal URLs

Closes #12.

### What this does
**Backend**
- `proposals` tables `proposal_id` is renamed `proposal_address`
- exposed `proposal_id`/`proposalId` is now db id
- migration: custom `alter_table` migration to rename the `proposal_id` col to `proposal_address` -- this was a little tricky to get it working with sqlite, not sure if we will encounter issues in prod env.
- tests updated
- bonus: exposed `update_id` on `proposal_update` model (FE was expecting this for a map->key)  

**Frontend**
- `Proposal.proposalId` is now of `number` type
- `Proposal.proposalAddress` has the eth address
- `Proposal.proposalUrlId` gets generated on consumption @ `api`
- `proposalUrlId` doesn't limit the number of words, I don't think this will be much of a problem once we validate title length, open to different opinions tho

### Testing
1. upgrade database
1. `flask test`
1. create a proposal > Publish and ensure the "click here" links to the readable url
1. browse proposals and make sure the links are the readable versions

### Screen
<img width="635" alt="screen shot 2018-11-06 at 5 34 34 pm" src="https://user-images.githubusercontent.com/11430954/48101669-73bc0200-e1ed-11e8-95f2-61f3ce4ed6e2.png">
